services/domains/domains.go

```diff
@@ -1,69 +1,31 @@
 package domains
 
 import (
-	"math"
-	"strings"
-
-	CoinType "github.com/trustwallet/blockatlas/coin"
 	"github.com/trustwallet/blockatlas/pkg/blockatlas"
 	"github.com/trustwallet/blockatlas/pkg/errors"
 	"github.com/trustwallet/blockatlas/platform"
 )
 
-// TLDMapping Mapping of name TLD's to coin where they are handled
-var TLDMapping = map[string]uint{
-	".eth":         CoinType.ETH,
-	".xyz":         CoinType.ETH,
-	".luxe":        CoinType.ETH,
-	".kred":        CoinType.ETH,
-	".zil":         CoinType.ZIL,
-	".crypto":      CoinType.ZIL,
-	"@trust":       CoinType.FIO,
-	"@trustwallet": CoinType.FIO,
-	"@binance":     CoinType.FIO,
-	"@fiomembers":  CoinType.FIO,
-	"@":            CoinType.FIO, // any FIO domain
-}
-
 func HandleLookup(name string, coins []uint64) ([]blockatlas.Resolved, error) {
-	// Assumption: format of the name can be decided (top-level-domain), and at most one naming service is tried
-	name = strings.ToLower(name)
-	tld, err := getTLD(name)
-	if err != nil {
-		return nil, errors.E(err, "name format not recognized", errors.Params{"name": name, "coins": coins})
-	}
-	id, ok := TLDMapping[tld]
-	if !ok {
-		// special handling for FIO, any fio domain
-		if len(tld) >= 2 && tld[0] == '@' {
-			tld = string("@")
-			id, ok = TLDMapping[tld]
-			if !ok {
-				return nil, errors.E("name not found", errors.Params{"name": name, "coins": coins, "tld": tld})
-			}
+	// Visit all providers, try lookup with all matching ones
+	// There must be at least one provider (normally ==1)
+	ret := []blockatlas.Resolved{}
+	matchingCount := 0
+	for _, api := range platform.NamingAPIs {
+		if !api.Match(name) {
+			continue
+		}
+		result, err := api.Lookup(coins, name)
+		if err != nil {
+			return nil, errors.E(err, "name format not recognized", errors.Params{"name": name, "coins": coins})
 		}
+		for _, r := range result {
+			ret = append(ret, r)
+		}
+		matchingCount++
```

Comment:
useless variable, we can handle logic with checking len(ret)

Reply:
Nope.  We want to check if any provider was asked, not that if the total responses for providers is empty.

Reply:
cannot see any value to do that

Reply:
If no provider can be identified, is a different error case than provider returns zero.  Was also like this before.

Reply:
If fio matches any domain - how it could be now?

Reply:
Fio matches any domain with @.

Reply:
Obtaining the provider from the name, and provider doing resolving are two distinct logical steps.  Errors from these two steps are kept distinct.